apps/arweave/src/ar_http_iface_server.erl

```diff
@@ -33,6 +36,21 @@
 %%%===================================================================
 %%% Public interface.
 %%%===================================================================
+start_link() ->
+	gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init(_) ->
+	?LOG_INFO(#{ start => ?MODULE, pid => self() }),
+	% this process needs to be stopped in a clean way,
+	% if something goes wrong, the connections must
+	% be cleaned before leaving.
+	erlang:process_flag(trap_exit, true),
+	{ok, Config} = application:get_env(arweave, config),
+	Semaphores = Config#config.semaphores,
+	case start_http_iface_listener(Config) of
```

Comment:
It is better to move long-lasting operations from `init/1` into `handle_continue/2`. Init callback blocks parent supervisor from starting other children.
Perhaps it is not critical in this case, but if theory it may take some time, and the whole application start will hang until this function returns.